src/main/java/it/gov/pagopa/payhub/activities/service/ingestionflow/treasury/csvcomplete/TreasuryCsvCompleteProcessingService.java

```diff
@@ -59,15 +61,40 @@ public TreasuryIufIngestionFlowFileResult processTreasuryCsvComplete(
     @Override
     protected boolean consumeRow(long lineNumber, TreasuryCsvCompleteIngestionFlowFileDTO row, TreasuryIufIngestionFlowFileResult ingestionFlowFileResult, List<TreasuryCsvCompleteErrorDTO> errorList, IngestionFlowFile ingestionFlowFile) {
         TreasuryCsvCompleteIngestionFlowFileResult treasuryCsvCompleteIngestionFlowFileResult = (TreasuryCsvCompleteIngestionFlowFileResult) ingestionFlowFileResult;
+        String ipa = treasuryCsvCompleteIngestionFlowFileResult.getIpaCode();
         try {
-            String ipa = treasuryCsvCompleteIngestionFlowFileResult.getIpaCode();
             if (!row.getOrganizationIpaCode().equalsIgnoreCase(ipa)) {
                 String errorMessage = String.format(
                         "Organization IPA code %s does not match with the one in the ingestion flow file %s",
                         row.getOrganizationIpaCode(), ipa);
                 log.error(errorMessage);
                 throw new OrganizationIpaCodeNotMatchException(errorMessage);
             }
+
+            Organization organization = organizationService.getOrganizationByIpaCode(ipa).orElseThrow(() -> {
+                String errorMessage = String.format("No organization found for IPA code %s", ipa);
+                log.error(errorMessage);
+                return new OrganizationIpaCodeNotMatchException(errorMessage);
+            });
```

Comment:
this cannot happen! at row 66 we are checking ig the IPA code match the ingestionFlowFile organization... thus it should already be checked...
if you need organizationId, add it to result as done for ipa code instead of calling the REST service for each row